matplotlibcpp.h

```diff
@@ -6,11 +6,16 @@
 #include <stdexcept>
 #include <iostream>
 
-#if __cplusplus > 199711L
+#if __cplusplus > 199711L || _MSC_VER > 1800
 #include <functional>
 #endif
 
-#include <python2.7/Python.h>
+#include <Python.h>
```

Comment:
I understand that you'd like to keep this solution as handy as possible and the current user support as it is now, but it says the support on python 2.7 will be done by 2020 and people are shifting to 3.X little by little.
I'm happy if this solution gets more flexiblity for choosing python version.

Is it possible to change the way of compiling something like following?

http://www.shocksolution.com/python-basics-tutorials-and-examples/linking-python-and-c-with-boostpython/

Otherwise I will explore to find preprocessor hacking that is supported at stdc++11 and probably will try to find python.h and otherwise python2.7/python.h. 
How do you think?